src/FondOfSpryker/Zed/SalesNewsletterSignup/Communication/Plugin/CheckoutRestApi/CustomerOptInOptOutUrlQuoteMapPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace FondOfSpryker\Zed\SalesNewsletterSignup\Communication\Plugin\CheckoutRestApi;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\RestCheckoutRequestAttributesTransfer;
+use Spryker\Zed\CheckoutRestApiExtension\Dependency\Plugin\QuoteMapperPluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Client\SalesNewsletterSignup\SalesNewsletterSignupFactory getFactory()
+ */
+class CustomerOptInOptOutUrlQuoteMapPlugin extends AbstractPlugin implements QuoteMapperPluginInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\RestCheckoutRequestAttributesTransfer $restCheckoutRequestAttributesTransfer
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function map(RestCheckoutRequestAttributesTransfer $restCheckoutRequestAttributesTransfer, QuoteTransfer $quoteTransfer): QuoteTransfer
```

Comment:
To me, the line seems very long.

Reply:
It's a little bit above 80.. think I will leave as is.